### ISSUE TITLE
Enum.value is accepted

### ISSUE THREAD

**Issue by erdnaxeli** [python/mypy#15340]
**Bug Report**

Mypy does not warn when using non existing attribute on an enum class.

**To Reproduce**

```python
from enum import Enum


class Test(Enum):
    TEST1 = "test1"
    TEST2 = "test2"


print(Test.TEST1.value)
print(Test.value)
```
https://mypy-play.net/?mypy=latest&python=3.11&gist=8e46b4e017e3f98bb9c4a2f31525eb8c

**Expected Behavior**

I expect mypy to warn me that `Test` has no attribute `value`:
```
error: "Type[Test]" has no attribute "value"  [attr-defined]
Found 1 error in 1 file (checked 1 source file)
```

**Actual Behavior**

Mypy see nothing wrong.

```
Success: no issues found in 1 source file
```

I suspect that because the `EnumType` class implements `__getattr__` mypy supposes any attribute can exist 🫤

**Your Environment**

- Mypy version used: latest
- Mypy command-line flags:
- Mypy configuration options from `mypy.ini` (and other config files):
- Python version used: 3.11

**Comment by erictraut** [comment]
The typeshed definition of the `Enum` class includes a `value` property, so mypy is doing the right thing from the perspective of the type stub. It would need to override the information in the stub and provide custom logic to detect this error.

**Comment by hauntsaninja** [comment]
Yeah, this is a won't fix. Note that if you try to use the value you'll usually get a type error since Test.value will be a Callable
